runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/wire/WireCoders.java

```diff
@@ -17,29 +17,65 @@
  */
 package org.apache.beam.runners.fnexecution.wire;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.IOException;
 import java.util.function.Predicate;
 import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.runners.core.construction.CoderTranslation;
 import org.apache.beam.runners.core.construction.ModelCoders;
+import org.apache.beam.runners.core.construction.RehydratedComponents;
 import org.apache.beam.runners.core.construction.SyntheticComponents;
 import org.apache.beam.runners.core.construction.graph.PipelineNode.PCollectionNode;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.util.WindowedValue.FullWindowedValueCoder;
 
 /** Helpers to construct coders for gRPC port reads and writes. */
 public class WireCoders {
-  /** Creates an SDK-side wire coder for a port read/write for the given PCollection. */
+  /**
+   * Creates an SDK-side wire coder for a port read/write for the given PCollection. Returns a
```

Comment:
"The returned coder will be a windowed value coder containing..."

Reply:
I've added a `@return` clause that documents this.